contracts/utils/Strings.sol

```diff
@@ -51,17 +51,18 @@ library Strings {
             string memory buffer = new string(length);
             /// @solidity memory-safe-assembly
             assembly {
-                let pos := add(buffer, 32)
-                let ptr := add(pos, length)
-
+                let ptr := add(buffer, add(32, length))
                 for {
 
-                } gt(ptr, pos) {
+                } 1 {
 
                 } {
                     ptr := sub(ptr, 1)
-                    mstore8(ptr, add(48, mod(value, 10)))
+                    mstore8(ptr, byte(mod(value, 10), _SYMBOLS))
```

Comment:
how is that compiled? Is the result cheaper than `add 48` I doubt it

Reply:
It is, I've measured it. Plus it's the same thing, have `value` and 2 constants on stack, perform one `mod` and one `Gverylow`operation.

Reply:
Fair enough.

Ideally we would need tooling that checks that for us.